packages/legend-studio-plugin-query-builder/src/stores/QueryResultSetModifierState.ts

```diff
@@ -179,7 +179,10 @@ export class QueryResultSetModifierState {
       val.parametersValues[1] = collection;
       currentExpression = val;
     }
-    if (this.limit || options?.overridingLimit) {
+    if (
+      this.canModifyLambdaInProjectionMode(lambda) &&
+      (this.limit || options?.overridingLimit)
+    ) {
```

Comment:
Is the best approach. 
What is the cost to do
graphFetch()->take()->serialize()
On large data sets, this would become an issue if we remove the take altogether

Reply:
`take()` simply does not work with `graphFetch` as I understood. The cost, therefore is tremendous - i.e. 💥